Build fails with boost 1.74.0

Hi,
The build fails with boost 1.74.0
The first error is
```
/tmp/icinga2/lib/base/utility.cpp:728:100: error: ‘fs::copy_option’ has not been declared
  728 |  fs::copy_file(fs::path(source.Begin(), source.End()), fs::path(target.Begin(), target.End()), fs::copy_option::overwrite_if_exists);
      |                                                                                                    ^~~~~~~~~~~
```
I am proposing a fix for this in https://github.com/Icinga/icinga2/pull/8184

However, I don't know how to fix the second error:
<details>

```
In file included from /usr/include/boost/smart_ptr/detail/sp_thread_sleep.hpp:22,
                 from /usr/include/boost/smart_ptr/detail/yield_k.hpp:23,
                 from /usr/include/boost/smart_ptr/detail/spinlock_gcc_atomic.hpp:14,
                 from /usr/include/boost/smart_ptr/detail/spinlock.hpp:42,
                 from /usr/include/boost/smart_ptr/detail/spinlock_pool.hpp:25,
                 from /usr/include/boost/smart_ptr/shared_ptr.hpp:29,
                 from /usr/include/boost/shared_ptr.hpp:17,
                 from /usr/include/boost/signals2/deconstruct.hpp:28,
                 from /usr/include/boost/signals2.hpp:15,
                 from /var/tmp/portage/net-analyzer/icinga2-2.12.0/work/icinga2-2.12.0_build/lib/remote/apilistener-ti.hpp:9,
                 from /var/tmp/portage/net-analyzer/icinga2-2.12.0/work/icinga2-2.12.0/lib/remote/apilistener.hpp:6,
                 from /var/tmp/portage/net-analyzer/icinga2-2.12.0/work/icinga2-2.12.0/lib/remote/apilistener.cpp:3:
/usr/include/boost/function_output_iterator.hpp:14:1: note: ‘#pragma message: This header is deprecated. Use <boost/iterator/function_output_iterator.hpp> instead.’
   14 | BOOST_HEADER_DEPRECATED("<boost/iterator/function_output_iterator.hpp>")
      | ^~~~~~~~~~~~~~~~~~~~~~~
/usr/include/boost/detail/scoped_enum_emulation.hpp:17:1: note: ‘#pragma message: This header is deprecated. Use <boost/core/scoped_enum.hpp> instead.’
   17 | BOOST_HEADER_DEPRECATED("<boost/core/scoped_enum.hpp>")
      | ^~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/boost/asio/spawn.hpp:21,
                 from /var/tmp/portage/net-analyzer/icinga2-2.12.0/work/icinga2-2.12.0/lib/base/io-engine.hpp:18,
                 from /var/tmp/portage/net-analyzer/icinga2-2.12.0/work/icinga2-2.12.0/lib/remote/jsonrpcconnection.hpp:8,
                 from /var/tmp/portage/net-analyzer/icinga2-2.12.0/work/icinga2-2.12.0/lib/remote/apilistener.hpp:7,
                 from /var/tmp/portage/net-analyzer/icinga2-2.12.0/work/icinga2-2.12.0/lib/remote/apilistener.cpp:3:
/usr/include/boost/asio/bind_executor.hpp: In instantiation of ‘boost::asio::detail::executor_binder_base<T, Executor, false>::executor_binder_base(E&&, U&&) [with E = boost::asio::strand<icinga::Shared<boost::asio::io_context::strand> >; U = void (* const&)(); T = void (*)(); Executor = boost::asio::execution::any_executor<boost::asio::execution::context_as_t<boost::asio::execution_context&>, boost::asio::execution::detail::blocking::never_t<0>, boost::asio::execution::prefer_only<boost::asio::execution::detail::blocking::possibly_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::tracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::untracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::fork_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::continuation_t<0> > >]’:
/usr/include/boost/asio/bind_executor.hpp:320:50:   required from ‘boost::asio::executor_binder<T, Executor>::executor_binder(const boost::asio::executor_binder<U, OtherExecutor>&) [with U = void (*)(); OtherExecutor = boost::asio::strand<icinga::Shared<boost::asio::io_context::strand> >; T = void (*)(); Executor = boost::asio::execution::any_executor<boost::asio::execution::context_as_t<boost::asio::execution_context&>, boost::asio::execution::detail::blocking::never_t<0>, boost::asio::execution::prefer_only<boost::asio::execution::detail::blocking::possibly_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::tracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::untracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::fork_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::continuation_t<0> > >]’
/usr/include/boost/asio/spawn.hpp:111:20:   required from ‘boost::asio::basic_yield_context<Handler>::basic_yield_context(const boost::asio::basic_yield_context<OtherHandler>&) [with OtherHandler = boost::asio::executor_binder<void (*)(), boost::asio::strand<icinga::Shared<boost::asio::io_context::strand> > >; Handler = boost::asio::executor_binder<void (*)(), boost::asio::execution::any_executor<boost::asio::execution::context_as_t<boost::asio::execution_context&>, boost::asio::execution::detail::blocking::never_t<0>, boost::asio::execution::prefer_only<boost::asio::execution::detail::blocking::possibly_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::tracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::untracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::fork_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::continuation_t<0> > > >]’
/usr/include/boost/asio/impl/spawn.hpp:355:24:   required from ‘void boost::asio::detail::coro_entry_point<Handler, Function>::operator()(typename boost::asio::basic_yield_context<Handler>::caller_type&) [with Handler = boost::asio::executor_binder<void (*)(), boost::asio::strand<icinga::Shared<boost::asio::io_context::strand> > >; Function = icinga::IoEngine::SpawnCoroutine<icinga::Shared<boost::asio::io_context::strand>, icinga::ApiListener::ListenerCoroutineProc(boost::asio::yield_context, const Ptr&, const Ptr&)::<lambda(boost::asio::yield_context)> >::<lambda(boost::asio::yield_context)>; typename boost::asio::basic_yield_context<Handler>::caller_type = boost::coroutines::pull_coroutine<void>]’
/usr/include/boost/coroutine/detail/push_coroutine_object.hpp:302:14:   required from ‘void boost::coroutines::detail::push_coroutine_object<PullCoro, void, Fn, StackAllocator>::run() [with PullCoro = boost::coroutines::pull_coroutine<void>; Fn = boost::asio::detail::coro_entry_point<boost::asio::executor_binder<void (*)(), boost::asio::strand<icinga::Shared<boost::asio::io_context::strand> > >, icinga::IoEngine::SpawnCoroutine<icinga::Shared<boost::asio::io_context::strand>, icinga::ApiListener::ListenerCoroutineProc(boost::asio::yield_context, const Ptr&, const Ptr&)::<lambda(boost::asio::yield_context)> >::<lambda(boost::asio::yield_context)> >&; StackAllocator = boost::coroutines::basic_standard_stack_allocator<boost::coroutines::stack_traits>]’
/usr/include/boost/coroutine/detail/trampoline_push.hpp:70:14:   required from ‘void boost::coroutines::detail::trampoline_push_void(boost::context::detail::transfer_t) [with Coro = boost::coroutines::detail::push_coroutine_object<boost::coroutines::pull_coroutine<void>, void, boost::asio::detail::coro_entry_point<boost::asio::executor_binder<void (*)(), boost::asio::strand<icinga::Shared<boost::asio::io_context::strand> > >, icinga::IoEngine::SpawnCoroutine<icinga::Shared<boost::asio::io_context::strand>, icinga::ApiListener::ListenerCoroutineProc(boost::asio::yield_context, const Ptr&, const Ptr&)::<lambda(boost::asio::yield_context)> >::<lambda(boost::asio::yield_context)> >&, boost::coroutines::basic_standard_stack_allocator<boost::coroutines::stack_traits> >]’
/usr/include/boost/coroutine/detail/push_coroutine_object.hpp:60:53:   [ skipping 13 instantiation contexts, use -ftemplate-backtrace-limit=0 to disable ]
/usr/include/boost/asio/impl/dispatch.hpp:211:49:   required from ‘typename boost::asio::async_result<typename std::decay<_Tp>::type, void()>::return_type boost::asio::dispatch(CompletionToken&&) [with CompletionToken = boost::asio::detail::spawn_helper<boost::asio::executor_binder<void (*)(), boost::asio::strand<icinga::Shared<boost::asio::io_context::strand> > >, icinga::IoEngine::SpawnCoroutine<icinga::Shared<boost::asio::io_context::strand>, icinga::ApiListener::ListenerCoroutineProc(boost::asio::yield_context, const Ptr&, const Ptr&)::<lambda(boost::asio::yield_context)> >::<lambda(boost::asio::yield_context)> >&; typename boost::asio::async_result<typename std::decay<_Tp>::type, void()>::return_type = void]’
/usr/include/boost/asio/impl/spawn.hpp:452:24:   required from ‘void boost::asio::spawn(Handler&&, Function&&, const boost::coroutines::attributes&, typename std::enable_if<(((! boost::asio::is_executor<typename std::decay<_Tp>::type>::value) && (! boost::asio::execution::is_executor<typename std::decay<_Tp>::type>::value)) && (! std::is_convertible<ExecutionContext&, boost::asio::execution_context&>::value))>::type*) [with Handler = boost::asio::executor_binder<void (*)(), boost::asio::strand<icinga::Shared<boost::asio::io_context::strand> > >; Function = icinga::IoEngine::SpawnCoroutine<icinga::Shared<boost::asio::io_context::strand>, icinga::ApiListener::ListenerCoroutineProc(boost::asio::yield_context, const Ptr&, const Ptr&)::<lambda(boost::asio::yield_context)> >::<lambda(boost::asio::yield_context)>; typename std::enable_if<(((! boost::asio::is_executor<typename std::decay<_Tp>::type>::value) && (! boost::asio::execution::is_executor<typename std::decay<_Tp>::type>::value)) && (! std::is_convertible<ExecutionContext&, boost::asio::execution_context&>::value))>::type = void]’
/usr/include/boost/asio/impl/spawn.hpp:491:21:   required from ‘void boost::asio::spawn(const boost::asio::strand<Executor>&, Function&&, const boost::coroutines::attributes&) [with Function = icinga::IoEngine::SpawnCoroutine<icinga::Shared<boost::asio::io_context::strand>, icinga::ApiListener::ListenerCoroutineProc(boost::asio::yield_context, const Ptr&, const Ptr&)::<lambda(boost::asio::yield_context)> >::<lambda(boost::asio::yield_context)>; Executor = icinga::Shared<boost::asio::io_context::strand>]’
/usr/include/boost/asio/impl/spawn.hpp:482:21:   required from ‘void boost::asio::spawn(const Executor&, Function&&, const boost::coroutines::attributes&, typename std::enable_if<(boost::asio::is_executor<Executor>::value || boost::asio::execution::is_executor<Executor>::value)>::type*) [with Function = icinga::IoEngine::SpawnCoroutine<icinga::Shared<boost::asio::io_context::strand>, icinga::ApiListener::ListenerCoroutineProc(boost::asio::yield_context, const Ptr&, const Ptr&)::<lambda(boost::asio::yield_context)> >::<lambda(boost::asio::yield_context)>; Executor = icinga::Shared<boost::asio::io_context::strand>; typename std::enable_if<(boost::asio::is_executor<Executor>::value || boost::asio::execution::is_executor<Executor>::value)>::type = void]’
/var/tmp/portage/net-analyzer/icinga2-2.12.0/work/icinga2-2.12.0/lib/base/io-engine.hpp:100:21:   required from ‘static void icinga::IoEngine::SpawnCoroutine(Handler&, Function) [with Handler = icinga::Shared<boost::asio::io_context::strand>; Function = icinga::ApiListener::ListenerCoroutineProc(boost::asio::yield_context, const Ptr&, const Ptr&)::<lambda(boost::asio::yield_context)>]’
/var/tmp/portage/net-analyzer/icinga2-2.12.0/work/icinga2-2.12.0/lib/remote/apilistener.cpp:441:150:   required from here
/usr/include/boost/asio/bind_executor.hpp:181:41: error: no matching function for call to ‘boost::asio::execution::any_executor<boost::asio::execution::context_as_t<boost::asio::execution_context&>, boost::asio::execution::detail::blocking::never_t<0>, boost::asio::execution::prefer_only<boost::asio::execution::detail::blocking::possibly_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::tracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::untracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::fork_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::continuation_t<0> > >::any_executor(boost::asio::strand<icinga::Shared<boost::asio::io_context::strand> >)’
  181 |       target_(BOOST_ASIO_MOVE_CAST(U)(u))
      |                                         ^
In file included from /usr/include/boost/asio/execution.hpp:19,
                 from /usr/include/boost/asio/any_io_executor.hpp:22,
                 from /usr/include/boost/asio/basic_deadline_timer.hpp:24,
                 from /usr/include/boost/asio/deadline_timer.hpp:24,
                 from /var/tmp/portage/net-analyzer/icinga2-2.12.0/work/icinga2-2.12.0/lib/base/io-engine.hpp:16,
                 from /var/tmp/portage/net-analyzer/icinga2-2.12.0/work/icinga2-2.12.0/lib/remote/jsonrpcconnection.hpp:8,
                 from /var/tmp/portage/net-analyzer/icinga2-2.12.0/work/icinga2-2.12.0/lib/remote/apilistener.hpp:7,
                 from /var/tmp/portage/net-analyzer/icinga2-2.12.0/work/icinga2-2.12.0/lib/remote/apilistener.cpp:3:
/usr/include/boost/asio/execution/any_executor.hpp:1409:3: note: candidate: ‘boost::asio::execution::any_executor<SupportableProperties>::any_executor(boost::asio::execution::any_executor<SupportableProperties>&&) [with SupportableProperties = {boost::asio::execution::context_as_t<boost::asio::execution_context&>, boost::asio::execution::detail::blocking::never_t<0>, boost::asio::execution::prefer_only<boost::asio::execution::detail::blocking::possibly_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::tracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::untracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::fork_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::continuation_t<0> >}]’
 1409 |   any_executor(any_executor&& other) BOOST_ASIO_NOEXCEPT
      |   ^~~~~~~~~~~~
/usr/include/boost/asio/execution/any_executor.hpp:1409:31: note:   no known conversion for argument 1 from ‘boost::asio::strand<icinga::Shared<boost::asio::io_context::strand> >’ to ‘boost::asio::execution::any_executor<boost::asio::execution::context_as_t<boost::asio::execution_context&>, boost::asio::execution::detail::blocking::never_t<0>, boost::asio::execution::prefer_only<boost::asio::execution::detail::blocking::possibly_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::tracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::untracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::fork_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::continuation_t<0> > >&&’
 1409 |   any_executor(any_executor&& other) BOOST_ASIO_NOEXCEPT
      |                ~~~~~~~~~~~~~~~^~~~~
/usr/include/boost/asio/execution/any_executor.hpp:1382:3: note: candidate: ‘boost::asio::execution::any_executor<SupportableProperties>::any_executor(const boost::asio::execution::any_executor<SupportableProperties>&) [with SupportableProperties = {boost::asio::execution::context_as_t<boost::asio::execution_context&>, boost::asio::execution::detail::blocking::never_t<0>, boost::asio::execution::prefer_only<boost::asio::execution::detail::blocking::possibly_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::tracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::untracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::fork_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::continuation_t<0> >}]’
 1382 |   any_executor(const any_executor& other) BOOST_ASIO_NOEXCEPT
      |   ^~~~~~~~~~~~
/usr/include/boost/asio/execution/any_executor.hpp:1382:36: note:   no known conversion for argument 1 from ‘boost::asio::strand<icinga::Shared<boost::asio::io_context::strand> >’ to ‘const boost::asio::execution::any_executor<boost::asio::execution::context_as_t<boost::asio::execution_context&>, boost::asio::execution::detail::blocking::never_t<0>, boost::asio::execution::prefer_only<boost::asio::execution::detail::blocking::possibly_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::tracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::untracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::fork_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::continuation_t<0> > >&’
 1382 |   any_executor(const any_executor& other) BOOST_ASIO_NOEXCEPT
      |                ~~~~~~~~~~~~~~~~~~~~^~~~~
/usr/include/boost/asio/execution/any_executor.hpp:1363:3: note: candidate: ‘template<class ... OtherSupportableProperties> boost::asio::execution::any_executor<SupportableProperties>::any_executor(boost::asio::execution::any_executor<OtherSupportableProperties ...>, typename std::enable_if<typename std::conditional<(! std::is_same<boost::asio::execution::any_executor<OtherSupportableProperties ...>, boost::asio::execution::any_executor<SupportableProperties> >::value), typename boost::asio::execution::detail::supportable_properties<0, void(SupportableProperties ...)>::is_valid_target<boost::asio::execution::any_executor<OtherSupportableProperties ...> >, std::integral_constant<bool, false> >::type::value>::type*) [with OtherSupportableProperties = {OtherSupportableProperties ...}; SupportableProperties = {boost::asio::execution::context_as_t<boost::asio::execution_context&>, boost::asio::execution::detail::blocking::never_t<0>, boost::asio::execution::prefer_only<boost::asio::execution::detail::blocking::possibly_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::tracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::untracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::fork_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::continuation_t<0> >}]’
 1363 |   any_executor(any_executor<OtherSupportableProperties...> other,
      |   ^~~~~~~~~~~~
/usr/include/boost/asio/execution/any_executor.hpp:1363:3: note:   template argument deduction/substitution failed:
In file included from /usr/include/boost/asio/spawn.hpp:21,
                 from /var/tmp/portage/net-analyzer/icinga2-2.12.0/work/icinga2-2.12.0/lib/base/io-engine.hpp:18,
                 from /var/tmp/portage/net-analyzer/icinga2-2.12.0/work/icinga2-2.12.0/lib/remote/jsonrpcconnection.hpp:8,
                 from /var/tmp/portage/net-analyzer/icinga2-2.12.0/work/icinga2-2.12.0/lib/remote/apilistener.hpp:7,
                 from /var/tmp/portage/net-analyzer/icinga2-2.12.0/work/icinga2-2.12.0/lib/remote/apilistener.cpp:3:
/usr/include/boost/asio/bind_executor.hpp:181:41: note:   ‘boost::asio::strand<icinga::Shared<boost::asio::io_context::strand> >’ is not derived from ‘boost::asio::execution::any_executor<OtherSupportableProperties ...>’
  181 |       target_(BOOST_ASIO_MOVE_CAST(U)(u))
      |                                         ^
In file included from /usr/include/boost/asio/execution.hpp:19,
                 from /usr/include/boost/asio/any_io_executor.hpp:22,
                 from /usr/include/boost/asio/basic_deadline_timer.hpp:24,
                 from /usr/include/boost/asio/deadline_timer.hpp:24,
                 from /var/tmp/portage/net-analyzer/icinga2-2.12.0/work/icinga2-2.12.0/lib/base/io-engine.hpp:16,
                 from /var/tmp/portage/net-analyzer/icinga2-2.12.0/work/icinga2-2.12.0/lib/remote/jsonrpcconnection.hpp:8,
                 from /var/tmp/portage/net-analyzer/icinga2-2.12.0/work/icinga2-2.12.0/lib/remote/apilistener.hpp:7,
                 from /var/tmp/portage/net-analyzer/icinga2-2.12.0/work/icinga2-2.12.0/lib/remote/apilistener.cpp:3:
/usr/include/boost/asio/execution/any_executor.hpp:1346:3: note: candidate: ‘template<class Executor> boost::asio::execution::any_executor<SupportableProperties>::any_executor(Executor, typename std::enable_if<typename std::conditional<((! std::is_same<Executor, boost::asio::execution::any_executor<SupportableProperties> >::value) && (! std::is_base_of<boost::asio::execution::detail::any_executor_base, Executor>::value)), boost::asio::execution::detail::is_valid_target_executor<Executor, void(SupportableProperties ...)>, std::integral_constant<bool, false> >::type::value>::type*) [with Executor = Executor; SupportableProperties = {boost::asio::execution::context_as_t<boost::asio::execution_context&>, boost::asio::execution::detail::blocking::never_t<0>, boost::asio::execution::prefer_only<boost::asio::execution::detail::blocking::possibly_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::tracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::untracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::fork_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::continuation_t<0> >}]’
 1346 |   any_executor(Executor ex,
      |   ^~~~~~~~~~~~
/usr/include/boost/asio/execution/any_executor.hpp:1346:3: note:   template argument deduction/substitution failed:
/usr/include/boost/asio/execution/any_executor.hpp: In substitution of ‘template<class Executor> boost::asio::execution::any_executor<boost::asio::execution::context_as_t<boost::asio::execution_context&>, boost::asio::execution::detail::blocking::never_t<0>, boost::asio::execution::prefer_only<boost::asio::execution::detail::blocking::possibly_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::tracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::untracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::fork_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::continuation_t<0> > >::any_executor(Executor, typename std::enable_if<typename std::conditional<((! std::is_same<Executor, boost::asio::execution::any_executor<boost::asio::execution::context_as_t<boost::asio::execution_context&>, boost::asio::execution::detail::blocking::never_t<0>, boost::asio::execution::prefer_only<boost::asio::execution::detail::blocking::possibly_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::tracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::untracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::fork_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::continuation_t<0> > > >::value) && (! std::is_base_of<boost::asio::execution::detail::any_executor_base, Executor>::value)), boost::asio::execution::detail::is_valid_target_executor<Executor, void(boost::asio::execution::context_as_t<boost::asio::execution_context&>, boost::asio::execution::detail::blocking::never_t<0>, boost::asio::execution::prefer_only<boost::asio::execution::detail::blocking::possibly_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::tracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::untracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::fork_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::continuation_t<0> >)>, std::integral_constant<bool, false> >::type::value, void>::type*) [with Executor = boost::asio::strand<icinga::Shared<boost::asio::io_context::strand> >]’:
/usr/include/boost/asio/bind_executor.hpp:181:41:   required from ‘boost::asio::detail::executor_binder_base<T, Executor, false>::executor_binder_base(E&&, U&&) [with E = boost::asio::strand<icinga::Shared<boost::asio::io_context::strand> >; U = void (* const&)(); T = void (*)(); Executor = boost::asio::execution::any_executor<boost::asio::execution::context_as_t<boost::asio::execution_context&>, boost::asio::execution::detail::blocking::never_t<0>, boost::asio::execution::prefer_only<boost::asio::execution::detail::blocking::possibly_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::tracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::untracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::fork_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::continuation_t<0> > >]’
/usr/include/boost/asio/bind_executor.hpp:320:50:   required from ‘boost::asio::executor_binder<T, Executor>::executor_binder(const boost::asio::executor_binder<U, OtherExecutor>&) [with U = void (*)(); OtherExecutor = boost::asio::strand<icinga::Shared<boost::asio::io_context::strand> >; T = void (*)(); Executor = boost::asio::execution::any_executor<boost::asio::execution::context_as_t<boost::asio::execution_context&>, boost::asio::execution::detail::blocking::never_t<0>, boost::asio::execution::prefer_only<boost::asio::execution::detail::blocking::possibly_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::tracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::untracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::fork_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::continuation_t<0> > >]’
/usr/include/boost/asio/spawn.hpp:111:20:   required from ‘boost::asio::basic_yield_context<Handler>::basic_yield_context(const boost::asio::basic_yield_context<OtherHandler>&) [with OtherHandler = boost::asio::executor_binder<void (*)(), boost::asio::strand<icinga::Shared<boost::asio::io_context::strand> > >; Handler = boost::asio::executor_binder<void (*)(), boost::asio::execution::any_executor<boost::asio::execution::context_as_t<boost::asio::execution_context&>, boost::asio::execution::detail::blocking::never_t<0>, boost::asio::execution::prefer_only<boost::asio::execution::detail::blocking::possibly_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::tracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::untracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::fork_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::continuation_t<0> > > >]’
/usr/include/boost/asio/impl/spawn.hpp:355:24:   required from ‘void boost::asio::detail::coro_entry_point<Handler, Function>::operator()(typename boost::asio::basic_yield_context<Handler>::caller_type&) [with Handler = boost::asio::executor_binder<void (*)(), boost::asio::strand<icinga::Shared<boost::asio::io_context::strand> > >; Function = icinga::IoEngine::SpawnCoroutine<icinga::Shared<boost::asio::io_context::strand>, icinga::ApiListener::ListenerCoroutineProc(boost::asio::yield_context, const Ptr&, const Ptr&)::<lambda(boost::asio::yield_context)> >::<lambda(boost::asio::yield_context)>; typename boost::asio::basic_yield_context<Handler>::caller_type = boost::coroutines::pull_coroutine<void>]’
/usr/include/boost/coroutine/detail/push_coroutine_object.hpp:302:14:   required from ‘void boost::coroutines::detail::push_coroutine_object<PullCoro, void, Fn, StackAllocator>::run() [with PullCoro = boost::coroutines::pull_coroutine<void>; Fn = boost::asio::detail::coro_entry_point<boost::asio::executor_binder<void (*)(), boost::asio::strand<icinga::Shared<boost::asio::io_context::strand> > >, icinga::IoEngine::SpawnCoroutine<icinga::Shared<boost::asio::io_context::strand>, icinga::ApiListener::ListenerCoroutineProc(boost::asio::yield_context, const Ptr&, const Ptr&)::<lambda(boost::asio::yield_context)> >::<lambda(boost::asio::yield_context)> >&; StackAllocator = boost::coroutines::basic_standard_stack_allocator<boost::coroutines::stack_traits>]’
/usr/include/boost/coroutine/detail/trampoline_push.hpp:70:14:   [ skipping 14 instantiation contexts, use -ftemplate-backtrace-limit=0 to disable ]
/usr/include/boost/asio/impl/dispatch.hpp:211:49:   required from ‘typename boost::asio::async_result<typename std::decay<_Tp>::type, void()>::return_type boost::asio::dispatch(CompletionToken&&) [with CompletionToken = boost::asio::detail::spawn_helper<boost::asio::executor_binder<void (*)(), boost::asio::strand<icinga::Shared<boost::asio::io_context::strand> > >, icinga::IoEngine::SpawnCoroutine<icinga::Shared<boost::asio::io_context::strand>, icinga::ApiListener::ListenerCoroutineProc(boost::asio::yield_context, const Ptr&, const Ptr&)::<lambda(boost::asio::yield_context)> >::<lambda(boost::asio::yield_context)> >&; typename boost::asio::async_result<typename std::decay<_Tp>::type, void()>::return_type = void]’
/usr/include/boost/asio/impl/spawn.hpp:452:24:   required from ‘void boost::asio::spawn(Handler&&, Function&&, const boost::coroutines::attributes&, typename std::enable_if<(((! boost::asio::is_executor<typename std::decay<_Tp>::type>::value) && (! boost::asio::execution::is_executor<typename std::decay<_Tp>::type>::value)) && (! std::is_convertible<ExecutionContext&, boost::asio::execution_context&>::value))>::type*) [with Handler = boost::asio::executor_binder<void (*)(), boost::asio::strand<icinga::Shared<boost::asio::io_context::strand> > >; Function = icinga::IoEngine::SpawnCoroutine<icinga::Shared<boost::asio::io_context::strand>, icinga::ApiListener::ListenerCoroutineProc(boost::asio::yield_context, const Ptr&, const Ptr&)::<lambda(boost::asio::yield_context)> >::<lambda(boost::asio::yield_context)>; typename std::enable_if<(((! boost::asio::is_executor<typename std::decay<_Tp>::type>::value) && (! boost::asio::execution::is_executor<typename std::decay<_Tp>::type>::value)) && (! std::is_convertible<ExecutionContext&, boost::asio::execution_context&>::value))>::type = void]’
/usr/include/boost/asio/impl/spawn.hpp:491:21:   required from ‘void boost::asio::spawn(const boost::asio::strand<Executor>&, Function&&, const boost::coroutines::attributes&) [with Function = icinga::IoEngine::SpawnCoroutine<icinga::Shared<boost::asio::io_context::strand>, icinga::ApiListener::ListenerCoroutineProc(boost::asio::yield_context, const Ptr&, const Ptr&)::<lambda(boost::asio::yield_context)> >::<lambda(boost::asio::yield_context)>; Executor = icinga::Shared<boost::asio::io_context::strand>]’
/usr/include/boost/asio/impl/spawn.hpp:482:21:   required from ‘void boost::asio::spawn(const Executor&, Function&&, const boost::coroutines::attributes&, typename std::enable_if<(boost::asio::is_executor<Executor>::value || boost::asio::execution::is_executor<Executor>::value)>::type*) [with Function = icinga::IoEngine::SpawnCoroutine<icinga::Shared<boost::asio::io_context::strand>, icinga::ApiListener::ListenerCoroutineProc(boost::asio::yield_context, const Ptr&, const Ptr&)::<lambda(boost::asio::yield_context)> >::<lambda(boost::asio::yield_context)>; Executor = icinga::Shared<boost::asio::io_context::strand>; typename std::enable_if<(boost::asio::is_executor<Executor>::value || boost::asio::execution::is_executor<Executor>::value)>::type = void]’
/var/tmp/portage/net-analyzer/icinga2-2.12.0/work/icinga2-2.12.0/lib/base/io-engine.hpp:100:21:   required from ‘static void icinga::IoEngine::SpawnCoroutine(Handler&, Function) [with Handler = icinga::Shared<boost::asio::io_context::strand>; Function = icinga::ApiListener::ListenerCoroutineProc(boost::asio::yield_context, const Ptr&, const Ptr&)::<lambda(boost::asio::yield_context)>]’
/var/tmp/portage/net-analyzer/icinga2-2.12.0/work/icinga2-2.12.0/lib/remote/apilistener.cpp:441:150:   required from here
/usr/include/boost/asio/execution/any_executor.hpp:1346:3: error: no type named ‘type’ in ‘struct std::enable_if<false, void>’
/usr/include/boost/asio/bind_executor.hpp: In instantiation of ‘boost::asio::detail::executor_binder_base<T, Executor, false>::executor_binder_base(E&&, U&&) [with E = boost::asio::strand<icinga::Shared<boost::asio::io_context::strand> >; U = void (* const&)(); T = void (*)(); Executor = boost::asio::execution::any_executor<boost::asio::execution::context_as_t<boost::asio::execution_context&>, boost::asio::execution::detail::blocking::never_t<0>, boost::asio::execution::prefer_only<boost::asio::execution::detail::blocking::possibly_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::tracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::untracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::fork_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::continuation_t<0> > >]’:
/usr/include/boost/asio/bind_executor.hpp:320:50:   required from ‘boost::asio::executor_binder<T, Executor>::executor_binder(const boost::asio::executor_binder<U, OtherExecutor>&) [with U = void (*)(); OtherExecutor = boost::asio::strand<icinga::Shared<boost::asio::io_context::strand> >; T = void (*)(); Executor = boost::asio::execution::any_executor<boost::asio::execution::context_as_t<boost::asio::execution_context&>, boost::asio::execution::detail::blocking::never_t<0>, boost::asio::execution::prefer_only<boost::asio::execution::detail::blocking::possibly_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::tracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::untracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::fork_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::continuation_t<0> > >]’
/usr/include/boost/asio/spawn.hpp:111:20:   required from ‘boost::asio::basic_yield_context<Handler>::basic_yield_context(const boost::asio::basic_yield_context<OtherHandler>&) [with OtherHandler = boost::asio::executor_binder<void (*)(), boost::asio::strand<icinga::Shared<boost::asio::io_context::strand> > >; Handler = boost::asio::executor_binder<void (*)(), boost::asio::execution::any_executor<boost::asio::execution::context_as_t<boost::asio::execution_context&>, boost::asio::execution::detail::blocking::never_t<0>, boost::asio::execution::prefer_only<boost::asio::execution::detail::blocking::possibly_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::tracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::untracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::fork_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::continuation_t<0> > > >]’
/usr/include/boost/asio/impl/spawn.hpp:355:24:   required from ‘void boost::asio::detail::coro_entry_point<Handler, Function>::operator()(typename boost::asio::basic_yield_context<Handler>::caller_type&) [with Handler = boost::asio::executor_binder<void (*)(), boost::asio::strand<icinga::Shared<boost::asio::io_context::strand> > >; Function = icinga::IoEngine::SpawnCoroutine<icinga::Shared<boost::asio::io_context::strand>, icinga::ApiListener::ListenerCoroutineProc(boost::asio::yield_context, const Ptr&, const Ptr&)::<lambda(boost::asio::yield_context)> >::<lambda(boost::asio::yield_context)>; typename boost::asio::basic_yield_context<Handler>::caller_type = boost::coroutines::pull_coroutine<void>]’
/usr/include/boost/coroutine/detail/push_coroutine_object.hpp:302:14:   required from ‘void boost::coroutines::detail::push_coroutine_object<PullCoro, void, Fn, StackAllocator>::run() [with PullCoro = boost::coroutines::pull_coroutine<void>; Fn = boost::asio::detail::coro_entry_point<boost::asio::executor_binder<void (*)(), boost::asio::strand<icinga::Shared<boost::asio::io_context::strand> > >, icinga::IoEngine::SpawnCoroutine<icinga::Shared<boost::asio::io_context::strand>, icinga::ApiListener::ListenerCoroutineProc(boost::asio::yield_context, const Ptr&, const Ptr&)::<lambda(boost::asio::yield_context)> >::<lambda(boost::asio::yield_context)> >&; StackAllocator = boost::coroutines::basic_standard_stack_allocator<boost::coroutines::stack_traits>]’
/usr/include/boost/coroutine/detail/trampoline_push.hpp:70:14:   required from ‘void boost::coroutines::detail::trampoline_push_void(boost::context::detail::transfer_t) [with Coro = boost::coroutines::detail::push_coroutine_object<boost::coroutines::pull_coroutine<void>, void, boost::asio::detail::coro_entry_point<boost::asio::executor_binder<void (*)(), boost::asio::strand<icinga::Shared<boost::asio::io_context::strand> > >, icinga::IoEngine::SpawnCoroutine<icinga::Shared<boost::asio::io_context::strand>, icinga::ApiListener::ListenerCoroutineProc(boost::asio::yield_context, const Ptr&, const Ptr&)::<lambda(boost::asio::yield_context)> >::<lambda(boost::asio::yield_context)> >&, boost::coroutines::basic_standard_stack_allocator<boost::coroutines::stack_traits> >]’
/usr/include/boost/coroutine/detail/push_coroutine_object.hpp:60:53:   [ skipping 13 instantiation contexts, use -ftemplate-backtrace-limit=0 to disable ]
/usr/include/boost/asio/impl/dispatch.hpp:211:49:   required from ‘typename boost::asio::async_result<typename std::decay<_Tp>::type, void()>::return_type boost::asio::dispatch(CompletionToken&&) [with CompletionToken = boost::asio::detail::spawn_helper<boost::asio::executor_binder<void (*)(), boost::asio::strand<icinga::Shared<boost::asio::io_context::strand> > >, icinga::IoEngine::SpawnCoroutine<icinga::Shared<boost::asio::io_context::strand>, icinga::ApiListener::ListenerCoroutineProc(boost::asio::yield_context, const Ptr&, const Ptr&)::<lambda(boost::asio::yield_context)> >::<lambda(boost::asio::yield_context)> >&; typename boost::asio::async_result<typename std::decay<_Tp>::type, void()>::return_type = void]’
/usr/include/boost/asio/impl/spawn.hpp:452:24:   required from ‘void boost::asio::spawn(Handler&&, Function&&, const boost::coroutines::attributes&, typename std::enable_if<(((! boost::asio::is_executor<typename std::decay<_Tp>::type>::value) && (! boost::asio::execution::is_executor<typename std::decay<_Tp>::type>::value)) && (! std::is_convertible<ExecutionContext&, boost::asio::execution_context&>::value))>::type*) [with Handler = boost::asio::executor_binder<void (*)(), boost::asio::strand<icinga::Shared<boost::asio::io_context::strand> > >; Function = icinga::IoEngine::SpawnCoroutine<icinga::Shared<boost::asio::io_context::strand>, icinga::ApiListener::ListenerCoroutineProc(boost::asio::yield_context, const Ptr&, const Ptr&)::<lambda(boost::asio::yield_context)> >::<lambda(boost::asio::yield_context)>; typename std::enable_if<(((! boost::asio::is_executor<typename std::decay<_Tp>::type>::value) && (! boost::asio::execution::is_executor<typename std::decay<_Tp>::type>::value)) && (! std::is_convertible<ExecutionContext&, boost::asio::execution_context&>::value))>::type = void]’
/usr/include/boost/asio/impl/spawn.hpp:491:21:   required from ‘void boost::asio::spawn(const boost::asio::strand<Executor>&, Function&&, const boost::coroutines::attributes&) [with Function = icinga::IoEngine::SpawnCoroutine<icinga::Shared<boost::asio::io_context::strand>, icinga::ApiListener::ListenerCoroutineProc(boost::asio::yield_context, const Ptr&, const Ptr&)::<lambda(boost::asio::yield_context)> >::<lambda(boost::asio::yield_context)>; Executor = icinga::Shared<boost::asio::io_context::strand>]’
/usr/include/boost/asio/impl/spawn.hpp:482:21:   required from ‘void boost::asio::spawn(const Executor&, Function&&, const boost::coroutines::attributes&, typename std::enable_if<(boost::asio::is_executor<Executor>::value || boost::asio::execution::is_executor<Executor>::value)>::type*) [with Function = icinga::IoEngine::SpawnCoroutine<icinga::Shared<boost::asio::io_context::strand>, icinga::ApiListener::ListenerCoroutineProc(boost::asio::yield_context, const Ptr&, const Ptr&)::<lambda(boost::asio::yield_context)> >::<lambda(boost::asio::yield_context)>; Executor = icinga::Shared<boost::asio::io_context::strand>; typename std::enable_if<(boost::asio::is_executor<Executor>::value || boost::asio::execution::is_executor<Executor>::value)>::type = void]’
/var/tmp/portage/net-analyzer/icinga2-2.12.0/work/icinga2-2.12.0/lib/base/io-engine.hpp:100:21:   required from ‘static void icinga::IoEngine::SpawnCoroutine(Handler&, Function) [with Handler = icinga::Shared<boost::asio::io_context::strand>; Function = icinga::ApiListener::ListenerCoroutineProc(boost::asio::yield_context, const Ptr&, const Ptr&)::<lambda(boost::asio::yield_context)>]’
/var/tmp/portage/net-analyzer/icinga2-2.12.0/work/icinga2-2.12.0/lib/remote/apilistener.cpp:441:150:   required from here
/usr/include/boost/asio/execution/any_executor.hpp:1339:3: note: candidate: ‘boost::asio::execution::any_executor<SupportableProperties>::any_executor(std::nullptr_t) [with SupportableProperties = {boost::asio::execution::context_as_t<boost::asio::execution_context&>, boost::asio::execution::detail::blocking::never_t<0>, boost::asio::execution::prefer_only<boost::asio::execution::detail::blocking::possibly_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::tracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::untracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::fork_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::continuation_t<0> >}; std::nullptr_t = std::nullptr_t]’
 1339 |   any_executor(nullptr_t) BOOST_ASIO_NOEXCEPT
      |   ^~~~~~~~~~~~
/usr/include/boost/asio/execution/any_executor.hpp:1339:16: note:   no known conversion for argument 1 from ‘boost::asio::strand<icinga::Shared<boost::asio::io_context::strand> >’ to ‘std::nullptr_t’
 1339 |   any_executor(nullptr_t) BOOST_ASIO_NOEXCEPT
      |                ^~~~~~~~~
/usr/include/boost/asio/execution/any_executor.hpp:1333:3: note: candidate: ‘boost::asio::execution::any_executor<SupportableProperties>::any_executor() [with SupportableProperties = {boost::asio::execution::context_as_t<boost::asio::execution_context&>, boost::asio::execution::detail::blocking::never_t<0>, boost::asio::execution::prefer_only<boost::asio::execution::detail::blocking::possibly_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::tracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::outstanding_work::untracked_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::fork_t<0> >, boost::asio::execution::prefer_only<boost::asio::execution::detail::relationship::continuation_t<0> >}]’
 1333 |   any_executor() BOOST_ASIO_NOEXCEPT
      |   ^~~~~~~~~~~~
/usr/include/boost/asio/execution/any_executor.hpp:1333:3: note:   candidate expects 0 arguments, 1 provided
make[2]: *** [lib/remote/CMakeFiles/remote.dir/build.make:156: lib/remote/CMakeFiles/remote.dir/apilistener.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1742: lib/remote/CMakeFiles/remote.dir/all] Error 2
make: *** [Makefile:185: all] Error 2
```

</details>

@Al2Klimov Is the referenced commit supposed to fix the build? I still get the same error after applying it
```
/var/tmp/portage/net-analyzer/icinga2-2.12.0/work/icinga2-2.12.0/lib/remote/jsonrpcconnection.cpp:55:112:   required from here
/usr/include/boost/asio/execution/any_executor.hpp:1346:3: error: no type named ‘type’ in ‘struct std::enable_if<false, void>’
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -172,6 +172,9 @@ add_definitions(-DBOOST_COROUTINES_NO_DEPRECATION_WARNING)
 
 add_definitions(-DBOOST_FILESYSTEM_NO_DEPRECATED)
 
+# Required for Boost v1.74+
+add_definitions(-DBOOST_ASIO_USE_TS_EXECUTOR_AS_DEFAULT)
+
 link_directories(${Boost_LIBRARY_DIRS})
 include_directories(${Boost_INCLUDE_DIRS})
 

